liquidity_pool/src/lib.rs

```diff
@@ -92,20 +81,20 @@ fn put_reserve_b(e: &Env, amount: i128) {
     e.storage().instance().set(&DataKey::ReserveB, &amount)
 }
 
-fn burn_shares(e: &Env, amount: i128) {
+fn burn_shares(e: &Env, from: &Address, amount: i128) {
+    let current_shares = get_shares(e, from);
+    if current_shares < amount {
+        panic!("insufficient shares");
+    }
     let total = get_total_shares(e);
-    let share_contract = get_token_share(e);
-
-    token::Client::new(e, &share_contract).burn(&e.current_contract_address(), &amount);
+    put_shares(e, from, current_shares - amount);
```

Comment:
There's no math safety around the modification of the current shares. Is that out of scope for the type of example this is trying to be?

Reply:
By "math safety" you mean a check that this is non negative? for `burn_shares()` or for `put_shares()`? `burn_shares()` panics if `current_shares` < `amount`

`put_total_shares()` in the current example doesn't include any checks, so I followed that pattern for `put_shares`

Reply:
Presumably that safety would be provided by the share token contract though.

Reply:
`put_total_shares()` in the current unmodified version is unchecked and does not touch the external token contract. See [here](https://github.com/stellar/soroban-examples/blob/9efe5065e644f4e4603d6a92cb6df5aa5508a483/liquidity_pool/src/lib.rs#L83).

But maybe I'm not understanding you. Can you clarify what check you'd like to see and where?

Reply:
I assumed there was overflow safety in the token impl, but actually there isn't, so I don't think anything is lost here.